proofs/tree_test.go

```diff
@@ -748,6 +748,13 @@ func TestDocumentTree_Generate_twice(t *testing.T) {
 	assert.EqualError(t, err, "tree already filled")
 }
 
+// Test DocumentTree sets rootHash correctly
+func TestDocumentTree_WithRootHash(t *testing.T) {
+	testRootHash := []byte{0xea, 0xa2, 0x2c, 0xc4, 0x1b, 0x91, 0x96, 0x23, 0x66, 0xc6, 0xa0, 0x8f, 0xaa, 0x49, 0xc0, 0xe8}
+	doctree := NewDocumentTreeWithRootHash(TreeOptions{Hash: sha256Hash, GetSalt: NewSaltForTest}, testRootHash)
+	assert.Equal(t, testRootHash, doctree.rootHash)
+}
```

Comment:
Can you add a validation of a merkle proof for this here? doctree.ValidateProof(...)

Reply:
Sure. Let me do that quickly and update the PR :)